base/special/exp.jl

```diff
@@ -136,3 +147,36 @@ function exp(x::T) where T<:Union{Float32,Float64}
         return T(1.0) - ((x*p)/(p - T(2.0)) - x)
     end
 end
+
+# Method
+# 1. Argument reduction: Reduce x to an r so that |r| <= ln(2)/512. Given x,
+#    find r and integers k, j such that
+#       x = k*ln(2) + j/256 + r,  0 <= j < 256, |r| <= ln(2)/512.
+#
+# 2. Approximate exp(r) by it's degree 3 taylor series around 0.
+#    Since the bounds on r are very tight, this is sufficient to be accurate to floating point epsilon.
+#
+# 3. Scale back: exp(x) = 2^k * 2^(j/256) * exp(r)
+#    Since the range of possible j is small, 2^(j/256) is simply stored for all possible values.
+#    Technichally the J_table stores a scaled version which makes subnormal numbers work better.
+
+
+function myexp(x::T) where T<:Float64
+    if !(abs(x) < abs(MAX_EXP(T)))
+        x <= -MIN_EXP(T) && return 0.
+        x >=  MAX_EXP(T) && return Inf
+        isnan(x) && return x
+    end
+    
+    N_float = muladd(x, LN2o256INV(T), MAGIC_ROUND_CONST(T))
+    N = reinterpret(Int64, N_float) % Int32
+    
+    N_float -=  MAGIC_ROUND_CONST(T)
+    r = muladd(N_float, LN2o256L(T), muladd(N_float, LN2o256U(T), x))
```

Comment:
I think you're assuming here that `muladd` will be a fused multiply-add (i.e. equivalent to `fma` and not separate multiply-add, otherwise the compensation assumptions won't work. 

Since `N_float` is an integer between `-275200 <= N_float <= 262144` it will have at most 19 bits set, so if you truncate `LN2o256U(T)` to 34 significant bits, and define `LN2o256L(T)` as the remainder it should then work.